src/Extension/Exception/ExtendingFailedException.php

```diff
@@ -0,0 +1,28 @@
+<?php
+
+/*
+ * This file is part of Flarum.
+ *
+ * For detailed copyright and license information, please view the
+ * LICENSE file that was distributed with this source code.
+ */
+
+namespace Flarum\Extension\Exception;
+
+use Exception;
+use Flarum\Extension\Extension;
+use Throwable;
+
+class ExtendingFailedException extends Exception
+{
+    public $extension;
+    public $extender;
+
+    public function __construct(Extension $extension, $extender, Throwable $previous = null)
+    {
+        $this->extension = $extension;
+        $this->extender = $extender;
+
+        parent::__construct($extension->getTitle().' has an invalid extend.php: '.get_class($extender), null, $previous);
```

Comment:
This message is a bit confusing. Maybe:

` has an invalid extend.php: error experienced when applying extender of type: `?

Reply:
Actually we might want to go a little further:

```
Experienced an error when booting extension: EXTENSION_TITLE
Error occured when applying an extender of type: EXTENDER_CLASS
```